src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -200,22 +200,33 @@ private void ResetCounters()
 
         private void OnTimer()
         {
-            Debug.Assert(Monitor.IsEntered(s_counterGroupLock));
             if (_eventSource.IsEnabled())
             {
-                DateTime now = DateTime.UtcNow;
-                TimeSpan elapsed = now - _timeStampSinceCollectionStarted;
+                DateTime now;
+                TimeSpan elapsed;
+                lock (s_counterGroupLock)
+                {
+                    now = DateTime.UtcNow;
+                    elapsed = now - _timeStampSinceCollectionStarted;
+                }
 
+                // MUST keep out of the scope of s_counterGroupLock because this will cause WritePayload
```

Comment:
I think ResetCounters() is protected by the lock, and the methods invoked in it is protected by the individual counters' lock so I don't think that would mess up the payload that's being written. It is possible for the value of a particular counter that is part of this group to be cleared out, although I don't think that would be a huge issue. Do you think otherwise?

Reply:
We chatted offline and concluded this fine as-is : )